doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py

```diff
@@ -3,16 +3,20 @@
 import numpy
 
 import Orange.data
-from Orange.widgets import widget, gui
+from Orange.widgets.widget import OWWidget, Input, Output
+from Orange.widgets import gui
 
-class OWDataSamplerA(widget.OWWidget):
+class OWDataSamplerA(OWWidget):
     name = "Data Sampler"
     description = "Randomly selects a subset of instances from the data set"
     icon = "icons/DataSamplerA.svg"
     priority = 10
 
-    inputs = [("Data", Orange.data.Table, "set_data")]
-    outputs = [("Sampled Data", Orange.data.Table)]
+    class Inputs:
+        data = Input("Data", Orange.data.Table)
+
+    class Outputs:
+        sample = Output("Sampled Data", Orange.data.Table)
```

Comment:
Sure, but this would be a meta-class of Inputs and Outputs, which are currently not derived from any other class - which I like, since its simpler to define them. Plus, as much as I like writing magic, I hate reading it, even if it's mine.

I suggest we don't include it in this PR, but, as we discussed today, perhaps make the same change for signals and messages, if we decide so.